crates/client/src/lib.rs

```diff
@@ -310,109 +316,164 @@ impl<R: RegistryStorage, C: ContentStorage, N: NamespaceMapStorage> Client<R, C,
-            }
-            Err(ClientError::PackageDoesNotExist {
-                name,
-                has_auth_token,
-            }) => {
-                if !initializing {
-                    let prompt = if has_auth_token {
-                        format!(
-"Package `{package_name}` does not already exist or you do not have access.
-Do you wish to initialize `{package_name}` and publish release y/N\n",
-package_name = &info.name,
-)
-                    } else {
-                        format!(
-"Package `{package_name}` does not already exist or you do not have access.
-You may be required to login. Try: `warg login`
-Do you wish to initialize `{package_name}` and publish release y/N\n",
-package_name = &info.name,
-)
-                    };
-                    if Confirm::with_theme(&ColorfulTheme::default())
-                        .with_prompt(prompt)
-                        .default(false)
-                        .interact()
-                        .unwrap()
-                    {
-                        info.entries.insert(0, PublishEntry::Init);
-                    } else {
+                Err(ClientError::PackageDoesNotExist {
+                    name,
+                    has_auth_token,
+                }) => {
+                    if !initializing && self.disable_interactive {
                         return Err(ClientError::MustInitializePackage {
                             name,
                             has_auth_token,
                         });
+                    } else if !initializing {
+                        #[cfg(feature = "cli-interactive")]
+                        {
+                            use crate::storage::PublishEntry;
+                            use dialoguer::{theme::ColorfulTheme, Confirm};
+
+                            if accepted_prompt_to_initialize
+                                || Confirm::with_theme(&ColorfulTheme::default())
+                                    .with_prompt(format!(
+                                        "Package `{package_name}` was not found.
+If it exists, you may not have access.
+Attempt to create `{package_name}` and publish the release y/N\n",
+                                        package_name = &info.name,
+                                    ))
+                                    .default(false)
+                                    .interact()
+                                    .unwrap()
+                            {
+                                info.entries.insert(0, PublishEntry::Init);
+                                initializing = true;
+                                accepted_prompt_to_initialize = true;
+                            } else {
+                                return Err(ClientError::MustInitializePackage {
+                                    name,
+                                    has_auth_token,
+                                });
+                            }
+                        }
```

Comment:
What happens here when the feature is disabled? Should it be returning `MustInitializePackage`?

Reply:
Revised to be cleaner and more correct.